handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -230,42 +230,44 @@ protected void deallocate() {
         this.jdkCompatibilityMode = jdkCompatibilityMode;
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
+        final long ssl;
         try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
+            this.ssl = ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
I don't think this is possible. The object may escape the constructor via the leak detector, but no methods should be invoked on the object being constructed as a result of this.